### ISSUE TITLE
'git flow release finish [VERSION]' appears to hang indefinitely

### ISSUE THREAD

**Issue by TAGC** [petervanderdoes/gitflow#175]
I should start by mentioning that I'm very new to using the Git flow system.

I'm using a 64-bit Windows 7 machine, and I attempted to merge a release-branch with master using the command `git flow release finish 2.0`. This created several merge conflicts, which I resolved and then committed with message 'Version 2.0'.

![](https://cloud.githubusercontent.com/assets/1588951/5261260/9d1a4886-7a0f-11e4-9c9a-1dc39d554b5b.PNG)

After that, I tried to continue the release completion process by running the command again: `git flow release finish 2.0`

However, at least using Git Bash, this appears to cause the terminal to hang indefinitely.

![](https://cloud.githubusercontent.com/assets/1588951/5261296/17010798-7a10-11e4-8cce-4f6bec57f0ae.PNG)

**Comment by TAGC** [comment]
Hello, sorry that it's been a while. I haven't had time to test this out at work yet, but I've noticed Git Bash hanging for other things unrelated to Git Flow e.g. 'git commit --amend'. It's probably just a Windows thing.

**Comment by gabeio** [comment]
@TAGC git can hang when you are running very low on ram with a very large repository because it has most of if not all of the repository in ram there's a few suggestions [here about large files](http://stackoverflow.com/questions/3055506/git-is-very-very-slow-when-tracking-large-binary-files) but they are mostly focusing on binary files. not sure if this will help but it might.

**Comment by TAGC** [comment]
@gabeio The computer at work has a decent amount of RAM (I think 4 GB) and the repositories I'm working with are pretty small. I noticed the 'git commit --amend' thing hang in a repository populated mainly with LabVIEW files which are each only a few dozen KB in size I think. Other Git commands run fine.

**Comment by gabeio** [comment]
huh weird :/ I guess not. just thought I would put it out there anyway. good luck finding the issue :smile:
Cheers!

**Comment by petervanderdoes** [comment]
@TAGC how many files are in the commit?

**Comment by TAGC** [comment]
@petervanderdoes I'll check tomorrow when I get to work, but I would estimate between 30-60.

**Comment by TAGC** [comment]
@petervanderdoes 

There are 68 files versioned in my HEAD commit.

The total size of my `.git` repository is 4.34 Mb.

I also get this information:

```
$ git count-objects -v
count: 331
size: 3622
in-pack: 207
packs: 1
size-pack: 793
prune-packable: 0
garbage: 0
size-garbage: 0
```

**Comment by petervanderdoes** [comment]
Not a real problem size wise then. I don't know Labview  but if you search for "Labview git" you do get quiet a few hits where people change their environment settings for that combination. 

I think it's more a combination of git and Labview than git-flow and Labview to be honest.

**Comment by TAGC** [comment]
@petervanderdoes I can't say for sure but I think I was getting this problem with Java applications as well. I vaguely remember trying to to 'git commit --amend' for a Java project, Git Bash hanging, so I did it instead through the EGit plugin for Eclipse.

**Comment by TAGC** [comment]
@petervanderdoes Quick update. I've mostly finished developing the LabVIEW project, but I wanted to break the project down into separate parts so that I can treat each part as an individual service and use them in other projects.

Since the functionality's already there, it meant I could quickly copy the files over, add a small amount of testing for good measure and call that 'version 1.0' of that service module. This gave me a good opportunity to retest the 'git flow release' problem I was having before.

Like before, I end up with Git Bash hanging, but this time it's managed to get partway through it seems. This is what the console output looks like currently:

```
davidfallah@DEVELOPMENT /w/davidfallah/Labview Projects/Message (release/1.0.0)
$ git flow release finish 1.0.0
C:/Program Files (x86)/Git/bin/gitflow-common: line 687: [: w:/davidfallah/Labview: binary operator expected
C:/Program Files (x86)/Git/bin/gitflow-common: line 715: [: w:/davidfallah/Labview: binary operator expected
Switched to branch 'master'
Merge made by the 'recursive' strategy.
 Message.aliases                                    |   3 +
 Message.lvlps                                      |   3 +
 Message.lvproj                                     |  23 ++++++
 .../labview/GenericMessage/GenericMessage.lvclass  |  83 +++++++++++++++++++++
 src/main/labview/GenericMessage/GetID.vi           | Bin 0 -> 23321 bytes
 src/main/labview/GenericMessage/GetPayload.vi      | Bin 0 -> 25518 bytes
 src/main/labview/GenericMessage/SetID.vi           | Bin 0 -> 23097 bytes
 src/main/labview/GenericMessage/SetPayload.vi      | Bin 0 -> 25454 bytes
 src/main/labview/GenericMessage/ToString.vi        | Bin 0 -> 36384 bytes
 .../labview/MessageFactory/CreateGenericMessage.vi | Bin 0 -> 34044 bytes
 .../labview/MessageFactory/MessageFactory.lvclass  |  35 +++++++++
 .../MessageFactory/CreateGenericMessageTest.lvtest | Bin 0 -> 2172 bytes
 .../MessageFactory/CreateGenericMessageTest.vi     | Bin 0 -> 26120 bytes
 13 files changed, 147 insertions(+)
 create mode 100644 Message.aliases
 create mode 100644 Message.lvlps
 create mode 100644 Message.lvproj
 create mode 100644 src/main/labview/GenericMessage/GenericMessage.lvclass
 create mode 100644 src/main/labview/GenericMessage/GetID.vi
 create mode 100644 src/main/labview/GenericMessage/GetPayload.vi
 create mode 100644 src/main/labview/GenericMessage/SetID.vi
 create mode 100644 src/main/labview/GenericMessage/SetPayload.vi
 create mode 100644 src/main/labview/GenericMessage/ToString.vi
 create mode 100644 src/main/labview/MessageFactory/CreateGenericMessage.vi
 create mode 100644 src/main/labview/MessageFactory/MessageFactory.lvclass
 create mode 100644 src/test/labview/lvtest/MessageFactory/CreateGenericMessageTest.lvtest
 create mode 100644 src/test/labview/testvi/MessageFactory/CreateGenericMessageTest.vi
```

The entire project folder is 700 KB. Git has about 13 files versioned. I'm using Git version 1.9.4.msysgit.1 for this.

You said earlier to try `set -x;git flow release finish 2.0` for another project. Trying it with this (for version 1.0.0), I just get:

```
davidfallah@DEVELOPMENT /w/davidfallah/Labview Projects/Message (release/1.0.0)
$ set -x;git flow release finish 1.0.0
+ git flow release finish 1.0.0
C:/Program Files (x86)/Git/bin/gitflow-common: line 687: [: w:/davidfallah/Labview: binary operator expected
C:/Program Files (x86)/Git/bin/gitflow-common: line 715: [: w:/davidfallah/Labview: binary operator expected
```

I've checked the specs for this computer: it's a 64-bit Windows 7 machine with 8 GB of RAM.

**Comment by petervanderdoes** [comment]
Thanks for the update:
1. The problem with `binary operator expected`.
What shell are you running and which version?  

I'm wondering, after you kill the process, is everything completed? Is the release branch deleted, is the master and develop branch the same?

**Comment by TAGC** [comment]
I'm using the Git Bash shell. I'm not which version.

One thing worth mentioning - I'm running Git Bash through the Mintty terminal emulator.

As a test, I tried creating two copies of the the project directory (which had a very small uncommitted change). I opened up two instances of Git Bash - one running through Mintty and one running through Window's native console. I made it so that each instance operated on one of the copies of the project directory and executed identical Git commands.

For each copy, I staged and committed the modified file and then began a 1.0.1 release. So far, both were working fine, although the one running through the native Windows console went a little faster. I then tried executing 'git flow release finish 1.0.1'. The native console one took a short while - and complained about a missing colour scheme in .vimrc if that's at all relevant - but it opened up Vim so I could enter in the tag message and then finished properly.

The one running through Mintty got stuck and is still stuck as I'm writing this.

After killing it, I find the following:
 - there are three branches (master, develop, release/1.0.1)
 - the master branch is checked out
 - the HEAD commit is "Merge branch 'release/1.0.1'"
 - no tag has been generated for v1.0.1

![Comparison](http://i.imgur.com/s0U9jee.png)
